models/issues/dependency.go

```diff
@@ -107,8 +107,8 @@ func (err ErrUnknownDependencyType) Unwrap() error {
 type IssueDependency struct {
 	ID           int64              `xorm:"pk autoincr"`
 	UserID       int64              `xorm:"NOT NULL"`
-	IssueID      int64              `xorm:"UNIQUE(issue_dependency) NOT NULL"`
-	DependencyID int64              `xorm:"UNIQUE(issue_dependency) NOT NULL"`
+	IssueID      int64              `xorm:"UNIQUE(issue_dependency) NOT NULL index"`
+	DependencyID int64              `xorm:"UNIQUE(issue_dependency) NOT NULL index"`
```

Comment:
Two nits:

1. Does it need a migration? For long time I have the question: 
    * If a migration is a must, it should be documented in guideline and explained.
    * If a migration is not a must, then we do not need to require to write migrations
2. The index for `IssueID` seems redundant, because `issue_dependency` should already provide the index for it.